### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by aintgoin2goa**
Use Basic Auth to secure api - with token coming via s3o.ft.com

### CODE REVIEW

**Comment by matthew-andrews** [comment]
is this useful https://github.com/Financial-Times/s3o-middleware ?

File: src/js/middleware/auth.js [hunk 1]
```diff
@@ -1,10 +1,59 @@
 'use strict';
+const crypto = require('crypto');
+const NodeRSA = require("node-rsa");
+const basicAuth = require('basic-auth');
+const debug = require('debug')('next-flags:authMiddleware');
 
 
+// Authenticate token and save/delete cookies as appropriate.
+// shamelessly copied and pasted from https://github.com/Financial-Times/S3O-middleware/
```
Comment (matthew-andrews) [inline src/js/middleware/auth.js]: ah just spotted this, ignore my comment

**Comment by aintgoin2goa** [comment]
Yeah - I copied and pasted the relevant bits

Will use the whole thing to protect the ui client (which will generate the token that the api simply validates)

**Comment by aintgoin2goa** [comment]
Phew!

**Pull request merged**